### PR TITLE
vimPlugins.sql-nvim: Set a default libsqlite3.so

### PR DESCRIPTION
**Pull request by jlesquembre**
###### Motivation for this change
The plugin relies on sqlite to work

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

ping @teto

### CODE REVIEW

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -287,8 +287,11 @@ self: super: {
   });
 
   sql-nvim = super.sql-nvim.overrideAttrs(old: {
-    buildInputs = [ sqlite ];
-  });
+    preFixup = ''
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:290]:
```suggestion
    postPatch = ''
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/misc/vim-plugins/overrides.nix (hunk 1 above)
Comment (jlesquembre) [reply pkgs/misc/vim-plugins/overrides.nix:290]: @SuperSandro2000 thanks for reviewing. I don't mind to do the change, the output will the same. But can you explain the reason for it? I thought that we should use the fixup phase for nix-specific post-processing actions, like in this case, setting the correct path to a .so file in the nix store.

**Review by teto — APPROVED** [review]
that fixes it for me. If you use this plugin for telescope/frecency, it would be nice to add sal-nvim as one of its buildinputs in vim-plugins/overrides.nix. Something I can also fix myself later.

**Comment by jlesquembre** [comment]
> that fixes it for me. If you use this plugin for telescope/frecency, it would be nice to add sal-nvim as one of its buildinputs in vim-plugins/overrides.nix. Something I can also fix myself later.

do you mean `dependecies` instead of `buildInputs`?

File: pkgs/misc/vim-plugins/overrides.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/misc/vim-plugins/overrides.nix:290]: fixup is usually used for fixing rpaths in binaries or similar. It does not make any difference here but code changes are usually done in patch.

**Comment by teto** [comment]
sry I was like "you should precisie it's not buildInputs" and I wrote `buildInputs` xD yes `dependencies`

File: pkgs/misc/vim-plugins/overrides.nix (hunk 1 above)
Comment (jlesquembre) [reply pkgs/misc/vim-plugins/overrides.nix]: @SuperSandro2000 thanks for clarifying it, I just updated it, I'll keep that in mind for future derivations

**Comment by jlesquembre** [comment]
@teto no worries, just wanted to be sure, I added the dependendy

**Comment by teto** [comment]
works for me thnks

**Pull request merged**